_posts/2023-10-24-filesystem-integrity-elytron-tool.adoc

```diff
@@ -0,0 +1,176 @@
+/subsystem=elytron/filesystem-realm=fsRealm:add-identity(identity=quickstartUser)
+/subsystem=elytron/filesystem-realm=fsRealm:set-password(identity=quickstartUser, digest={password=password123!, realm=fsRealm, algorithm=digest-md5})
+/subsystem=elytron/filesystem-realm=fsRealm:add-identity-attribute(identity=quickstartUser, name=Roles, value=["Admin", "Guest"])
+``` 
+Now if we navigate to the server configuration directory, we will see a new directory created, called `fs-realm`, which contains files relevant to the filesystem realm we just created. If you navigate to the xml file inside the folder, you will see that the details related to the identity we added are all human readable. Additionally, the name of the principal appears in the name of the xml file. An example of the contents of the xml file for a filesystem without integrity checking can be seen below: 
+
+```
+<principal name="quickstartUser"/>
+    <credentials>
+        <password algorithm="digest-md5" format="base64">Ag5xdWlja3N0YXJ0VXNlcgdmc1JlYWxt4Y4F7KmGKSCPrtzjzHD8Hg==</password>
+    </credentials>
+    <attributes>
+        <attribute name="Roles" value="Admin"/>
+        <attribute name="Roles" value="Guest"/>
+    </attributes>
+```
+
+We then create a new security domain which will make use of our filesystem realm as follows:
+```
+/subsystem=elytron/security-domain=fsDomain:add(realms=[{realm=fsRealm}], default-realm=fsRealm,permission-mapper=default-permission-mapper)
+```
+And lastly, update our security domain mapping in the Undertow subsystem:
+```
+/subsystem=undertow/application-security-domain=other:write-attribute(name=security-domain, value=fsDomain)
+```
+Now we can deploy and access the application and log in using the credentials for quickstartUser. However, without any integrity checking, anyone with access to the contents of the filesystem realm can tamper with it. So, let's add integrity checking for this filesystem realm. 
+
+== Adding Integrity Checking
+In order to access the elytron tool, we can use the following command: 
+```
+./bin/elytron-tool.sh filesystem-realm-integrity --input-location WILDFLY_HOME/standalone/configuration/fs-realm --output-location WILDFLY_HOME/standalone/configuration/fsRealmWithIntegrity  --keystore PATH/TO/KEYSTORE/FILE --password password --key-pair key --summary
+```
+
+Here the `input-location` option refers to the path to the filesystem realm directory, the `output-location` refers to the new filesystem realm to be created. The `keystore` option specifies the location for the keystore file, `password` refers to the password for the keystore and `key-pair` refers to the alias of the key that uniquely identified the key pair we are looking for inside the keystore. The `summary` option is added so that when the new filesystem is created, we can see the details of what is being done. The keystore can be created in one of two ways: 
```

Comment:
`input-location` option refers to the path to the existing filesystem

Reply:
s/identified/identifies

Reply:
fixed!